### PR TITLE
fsr to sr

### PR DESCRIPTION
**Pull request by samuelshaner**
In preparation for PR #273 in implementing the linear source approximation, this PR changes fsr to sr throughout the repository and removes the mention of "flat", where appropriate.

### CODE REVIEW

**Comment by geogunow** [comment]
Looks good to me, though I only did a quick skim

File: docs/source/methods/method_of_characteristics.rst [hunk 1]
```diff
@@ -18,9 +18,9 @@ The method of characteristics (MOC) is a widely used technique for solving parti
 * track area approximation (:ref:`Section 2.9 <track-area-approximation>`)
 * azimuthal planar projection (:ref:`Section 2.10 <projection-azimuthal-plane>`)
 
-The final equations applied in OpenMOC to solve for the FSR source and scalar flux derived in the following sections are summarized below:
+The final equations applied in OpenMOC to solve for the SR source and scalar flux derived in the following sections are summarized below:
 
-**The source in each flat source region**
+**The source in each source region**
```
Comment (wbinventor) [inline docs/source/methods/method_of_characteristics.rst]: These equations relate to the "flat" source region approximation, so should we not retain that terminology here? Are you thinking of breaking this up into two sections for fixed and linear source approximations?

File: docs/source/methods/method_of_characteristics.rst [hunk 2]
```diff
@@ -261,7 +261,7 @@ The subscripts :math:`m` and :math:`p` for the azimuthal and polar angles, respe
 The Flat Source Region Approximation
 ====================================
 
-Another common approximation for MOC is to assume that the source :math:`Q_g` is constant across discrete spatial cells termed *flat source regions* (FSRs). This implies that the source does not vary along a characteristic :math:`k` entering FSR :math:`i` at :math:`s'` and exiting at :math:`s''`: 
+Another common approximation for MOC is to assume that the source :math:`Q_g` is constant across discrete spatial cells termed *source regions* (SRs). This implies that the source does not vary along a characteristic :math:`k` entering FSR :math:`i` at :math:`s'` and exiting at :math:`s''`:
```
Comment (wbinventor) [inline docs/source/methods/method_of_characteristics.rst]: In this case I think that "flat source region" is appropriate based on the title of the section

File: docs/source/methods/method_of_characteristics.rst [hunk 3]
```diff
@@ -274,7 +274,7 @@ Another common approximation for MOC is to assume that the source :math:`Q_g` is
 The Constant Cross-Section Approximation
 ========================================
 
-In addition to the flat source approximation, it is assumed that the material properties are constant across each FSR. The area-averaged cross-sections for FSR :math:`i \in \{1, 2, ..., I\}` with area :math:`A_{i}` are defined as:
+In addition to the flat source approximation, it is assumed that the material properties are constant across each SR. The area-averaged cross-sections for SR :math:`i \in \{1, 2, ..., I\}` with area :math:`A_{i}` are defined as:
```
Comment (wbinventor) [inline docs/source/methods/method_of_characteristics.rst]: I guess I'm not really sure how you want to refactor this documentation, but it appears you may have searched for "flat source region" but may also want to look for "flat source approximation," if we deem it necessary to change the terminology here (or instead add a second section on linear source approximation).

Comment (wbinventor) [inline docs/source/methods/method_of_characteristics.rst]: Or based on what I see below, look for simply "flat source"

File: docs/source/usersguide/processing.rst [hunk 4]
```diff
@@ -425,10 +425,10 @@ A depiction of the cells for the :file:`/OpenMOC/sample-input/simple-lattice.py`
 .. note:: The runtime required by the plotting routine scales with the number of pixels in the image (the square of the ``gridsize`` parameter).
 
 
-Plotting by FSR
----------------
+Plotting by SR
+--------------
 
-To plot the geometry color-coded by the flat source region ID's throughout the geometry, use the ``plot_flat_source_regions(...)`` routine in the ``openmoc.plotter`` module. The parameters accepted by this routine are described in :ref:`Table 12 <table_plot_fsrs>`.
+To plot the geometry color-coded by the Source region ID's throughout the geometry, use the ``plot_source_regions(...)`` routine in the ``openmoc.plotter`` module. The parameters accepted by this routine are described in :ref:`Table 12 <table_plot_fsrs>`.
```
Comment (wbinventor) [inline docs/source/usersguide/processing.rst]: I don't think "Source" needs to be capitalized here

File: openmoc/plotter.py [hunk 5]
```diff
@@ -1015,10 +1015,10 @@ def plot_spatial_data(domains_to_data, plot_params, get_figure=False):
     Examples
     --------
     A user may invoke this function from an Python file as follows:
```
Comment (wbinventor) [inline openmoc/plotter.py]: Can you fix this typo from before here and elsewhere: "...from an Python file" -> "...from a Python file..."

File: src/CPUSolver.cpp [hunk 6]
```diff
@@ -659,8 +659,8 @@ void CPUSolver::transportSweep() {
   int min_track = 0;
   int max_track = 0;
 
-  /* Initialize flux in each FSr to zero */
-  flattenFSRFluxes(0.0);
+  /* Initialize flux in each Sr to zero */
```
Comment (wbinventor) [inline src/CPUSolver.cpp]: This should be "SR" rather than "Sr"

File: src/CPUSolver.cpp [hunk 7]
```diff
@@ -859,40 +859,40 @@ void CPUSolver::addSourceToScalarFlux() {
...
   FP_PRECISION* sigma_f;
   FP_PRECISION volume;
 
   /* Initialize fission rates to zero */
-  for (int r=0; r < _num_FSRs; r++)
+  for (int r=0; r < _num_SRs; r++)
     fission_rates[r] = 0.0;
 
-  /* Loop over all FSRs and compute the volume-averaged fission rate */
+  /* Loop over all SRs and compute the volume-averaged nu-fission rate */
```
Comment (wbinventor) [inline src/CPUSolver.cpp]: Actually this is now computing the volume-averaged "fission" rate rather than the "nu-fission" rate

Comment (wbinventor) [inline src/CPUSolver.cpp]: (as it should be)

File: docs/source/methods/method_of_characteristics.rst (hunk 1 above)
Comment (samuelshaner) [inline docs/source/methods/method_of_characteristics.rst]: Reading this section again, it is probably appropriate to retain the use of "flat" source and then I'll add another section on the linear source approximation. I'll add flat back in here.

File: docs/source/methods/method_of_characteristics.rst (hunk 2 above)
Comment (samuelshaner) [inline docs/source/methods/method_of_characteristics.rst]: I agree - I'll change it back.

File: docs/source/usersguide/processing.rst (hunk 4 above)
Comment (samuelshaner) [inline docs/source/usersguide/processing.rst]: Good catch, I'll un-capitalize it.

File: openmoc/plotter.py (hunk 5 above)
Comment (samuelshaner) [inline openmoc/plotter.py]: Sure. If in looking through this you find an other issues feel free to point them out and I'll be sure to update them.

File: src/CPUSolver.cpp (hunk 6 above)
Comment (samuelshaner) [inline src/CPUSolver.cpp]: Good catch!

File: src/CPUSolver.cpp (hunk 7 above)
Comment (samuelshaner) [inline src/CPUSolver.cpp]: Indeed. Glad you pointed out this detail of the compute fission sources procedure out!

**Comment by wbinventor** [comment]
Whew, reviewing this PR was a wake up call to just how much of our code revolved around the idea of "flat source regions." Everything looks good save for my few minor comments. What do you and @geogunow think about waiting to merge this PR in until a) v0.2 is released and b) we have resolved any known issues with #273? I'd prefer to make sure that this and #273 are merged nearly back-to-back so there isn't a long lasting gap in our terminology and implementation.

**Comment by wbinventor** [comment]
Also, I do want to test out all if the sample input files with this code before merging just a second sanity check. But that can wait till linear source matures a bit more.

**Comment by samuelshaner** [comment]
Sounds good to me. The unresolved items in #273 will probably take at least another week to get implemented and I might even parcel some of those items out into separate PRs, if possible, to keep things manageable.

**Comment by gridley** [comment]
Closing due to inactivity. Sorry Sam! 😃

**Pull request closed without merging**